lib/parse-message.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+module.exports = exports = {};
+
+exports.nickCommand = function (message, socket) {
+  socket.username = message.split(' ').slice(1).join(' ').trim();
+  socket.write(`you are now ${socket.username}\n`);
+};
+
+exports.dmCommand = function dmCommand(message, array){
+  // "/dm user_30 hello how are you"
+  let toUsername = message.split(' ')[1];
+  let content = message.split(' ').slice(2).join(' ').trim();
+  array.forEach(s => {
+    if (s.username === toUsername)
+      s.write(content);
+  });
+};
+
+exports.userCommand = function userCommand(array, socket){
+  array.forEach(user => {
+    socket.write(user.username + '\n');
+  });
+};
+
+exports.trollCommand = function(message, array){
+  let content = message.split(' ').slice(1).join(' ').trim();
+  array.forEach(user => {
+    for (var i = 0; i < 10; i++) {
+      user.write(content  + '\n');
+    }
+  });
+};
+
+exports.banCommand = function(message, array){
+  let toUsername = message.split(' ')[1].trim();
+  array.forEach(s => {
+    if (s.username === toUsername)
+      s.end(`${s.username} has been banned `);
```

Comment:
Good job with figuring out the ``.end()`` method, but remember that you have delete the user from your sockets array too. Duncan does that already in his server file when a user closes their own connection, but you should bring that same logic to this function when someone gets kicked out.